src/main/java/com/redhat/devtools/intellij/common/utils/DownloadHelper.java

```diff
@@ -138,7 +144,7 @@ private CompletableFuture<ToolInstance> downloadIfRequiredAsyncInner(String tool
             Path path = Paths.get(tool.getBaseDir().replace("$HOME", CommonConstants.HOME_FOLDER), "cache", tool.getVersion(), command);
             final String cmd = path.toString();
             if (!Files.exists(path)) {
-                downloadInBackground(toolName, platform, path, cmd, tool, version, result);
+                return downloadInBackground(toolName, platform, path, cmd, tool, version, platform.getSha256());
```

Comment:
a return in the middle of a big block which sets up a return value initially and returns it in the end looks a bit odd. 
I somehow preferred the prior version where the return value (`result`) was passed into the method rather than creating a new one. Matter of gusto.

Reply:
you're right, i might simply use the `result` instance here